module/Company/src/Company/Controller/AdminController.php

```diff
@@ -0,0 +1,345 @@
+                                                           array('slugCompanyName' => $companyName))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $packetForm,
+        ));
+        
+        return $vm;
+        
+    }
+    public function addCompanyAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyForm = $companyService->getCompanyForm();
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($companyName)){
+                $companyName = $request->getPost()['slugName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm = $companyService->getCompanyForm();
+            $companyForm->setData($request->getPost());
+
+            // TODO: isValid does not work yet
+            //if ($companyForm->isValid()) {
+            //    $company = $companyService->insertCompany();
+            //    $company->exchangeArray($request->getPost()); // Temporary fix, bind does not work yet?
+            //    $companyService->saveCompany();
+            //    return $this->redirect()->toRoute('admin_company/default', 
+             //                                     array('action'=>'edit', 
+             //                                           'slugCompanyName' => $companyName), 
+             //                                     array(), false);   
+            //}
+            if ($companyForm->isValid()) {
+                $company=$companyService->insertCompany($request->getPost()["languages"]);
+                $company->exchangeArray($request->getPost()); 
+                $companyService->saveCompany();
+                //var_dump($company);
+                return $this->redirect()->toRoute('admin_company/default', array('action'=>'edit', 'slugCompanyName'=>$companyName),array(),false);   
+            }
+        }
+        //$company = $companyService->insertCompany();
+        //$companyForm->bind($company);
+        $companyForm->setAttribute('action', 
+                                   $this->url()->fromRoute('admin_company/default', 
+                                                           array('action' => 'addCompany'))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $companyForm,
+        ));
+        
+        return $vm;
+        
+    }
+    public function deletePacketAction(){
+        $companyService = $this->getCompanyService();
+        $packetID = $this->params('packetID');    
+        $companyName = $this->params('slugCompanyName');
+        $request = $this->getRequest();
+         if ($request->isPost()) {
+             $del = $request->getPost('del', 'No');
+             if ($del == 'Yes') {
```

Comment:
Also, if a POST request is being made to a route named `/delete` then why does there need to be a parameter `del=Yes|No` which specifies if something should be deleted or not?